py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -104,16 +104,46 @@ def __init__(
         self.profiling_enabled = settings.debug if settings.debug is not None else False
         self.settings = settings
         self.engine = None
+        self.context = None
         self.weight_name_map = weight_name_map
 
         if self.serialized_engine is not None and not self.settings.lazy_engine_init:
             self.setup_engine()
 
+    def reset_context(self):
+        # if runtime configuration has modified, the context needs to be recreated in next forward() or enable_profiling()
+        if self.context is not None:
+            del self.context
+            self.context = None
+
+    def init_context(self):
+        assert self.engine, f"Context is used before setting up the engine"
+        if self.context is None:
+            self.context = self.engine.create_execution_context()
+
+    def get_weight_streaming_budget(self):
+        return self.engine.streamable_weights_size
+
+    def set_weight_streaming_budget(self, budget_bytes):
+        self.reset_context()
+        self.engine.weight_streaming_budget_v2 = budget_bytes
+        if self.engine.weight_streaming_budget_v2 != budget_bytes:
+            logger.error(f"Failed to set weight streaming budget to {budget_bytes}")
+            budget_bytes = self.engine.weight_streaming_budget_v2
+        if self.engine.streamable_weights_size == budget_bytes:
+            logger.warning("Weight streaming is disabled")
+
+        return budget_bytes
+
+    def set_automatic_streaming_budget(self):
+        budget_bytes = self.engine.get_weight_streaming_automatic_budget()
+        return self.set_weight_streaming_budget(budget_bytes)
+
```

Comment:
This api is same as in [TorchTensorRTModule](https://github.com/pytorch/TensorRT/pull/3111/commits/2fe93697e9d7274968cc639ee51a9e360135dcac#diff-10ebe63ef48658946436e5c01d9b062c2527b2a8990a6b6ab8b067c1783571baR137-R154) class. If this interface is good to go, parent class can used to share it and other some methods.